include/ensmallen_bits/pso/pso_impl.hpp

```diff
@@ -150,7 +158,9 @@ double PSOType<VelocityUpdatePolicy, InitPolicy>::Optimize(
     // In-place update of particle positions.
     particlePositions += particleVelocities;
 
+    //TODO: Think over this. Probably should not be parallelized.
     //Find the best particle.
+    #pragma omp parallel for
```

Comment:
Perhaps it makes sense to calculate ``bestFitness`` afterwards, in an independent run, that way there is no need to synchronize between different runs.

Reply:
What here need is a reduction clause. Since here you essentially need an argmin, there are 2 strategies to do this. The easier way is simply use
```suggestion
    #pragma omp parallel for reduction(min:bestFitness)
```
You can find the `bestParticle` later after the parallelized region. `reduction` is supported in OpenMP since 3.0.

Another way is to define a custom reduction to calculate `bestParticle` and `bestFitness` in one pass by using `declare reduction` directive introduced in OpenMP 4.0. See section 2.15 of https://www.openmp.org/wp-content/uploads/OpenMP4.0.0.pdf

Reply:
Hey @SuryodayBasak, just checking in---do you think you'll have a chance to handle this?  I think @zhlou's suggestion here is a good one.  It's ok if not, of course; I just want to make sure it's still on the radar if you are interested in finishing it up and getting it merged. :smile: